rcp/org.testeditor.rcp4/src/org/testeditor/rcp4/Application.xtend

```diff
@@ -12,22 +12,37 @@
  *******************************************************************************/
 package org.testeditor.rcp4;
 
+import java.io.IOException
+import javax.swing.JOptionPane
 import org.eclipse.equinox.app.IApplication
 import org.eclipse.equinox.app.IApplicationContext
+import org.eclipse.swt.SWT
+import org.eclipse.swt.widgets.Button
+import org.eclipse.swt.widgets.Event
+import org.eclipse.swt.widgets.Listener
+import org.eclipse.swt.widgets.MessageBox
+import org.eclipse.swt.widgets.Shell
 import org.eclipse.ui.PlatformUI
 import org.slf4j.LoggerFactory
 
 /**
  * This class controls all aspects of the application's execution
  */
 public class Application implements IApplication {
-	static val logger=LoggerFactory.getLogger(Application)
+	static val logger = LoggerFactory.getLogger(Application)
 
 	/* (non-Javadoc)
 	 * @see IApplication#start(org.eclipse.equinox.app.IApplicationContext)
 	 */
 	override start(IApplicationContext context) {
 		logger.info("STARTING APPLICATION ..")
+		try {
+			Runtime.getRuntime().exec(#{"javac"})
+		} catch (IOException e) {
+			if(JOptionPane.showConfirmDialog(null, "Application is started with a jre. For test-execution a jdk is needed. Start anyway?")!= 0) {
```

Comment:
You will not be able to run any test if you start the testeditor with a jre. I think we have to inform the user about this limitation.